tests/Unit/NumericalAlgorithms/LinearOperators/Test_PartialDerivatives.cpp

```diff
@@ -156,6 +157,35 @@ void test_logical_partial_derivative_per_tensor(
   });
 }
 
+template <typename GradientTags, typename VariableTags, size_t Dim,
+          typename DerivativeFrame>
+void test_partial_derivative_per_tensor(
+    const Variables<GradientTags>& du, const Variables<VariableTags>& u,
+    const Mesh<Dim>& mesh,
+    const InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                          DerivativeFrame>& inverse_jacobian) {
+  tmpl::for_each<GradientTags>([&du, &mesh, &u,
+                                &inverse_jacobian](auto gradient_tag_v) {
+    using gradient_tag = tmpl::type_from<decltype(gradient_tag_v)>;
+    using var_tag = typename gradient_tag::tag;
+
+    const auto single_du =
+        partial_derivative(get<var_tag>(u), mesh, inverse_jacobian);
+
+    Approx local_approx = Approx::custom().epsilon(1e-14).scale(1.0);
+    CHECK_ITERABLE_CUSTOM_APPROX(single_du, get<gradient_tag>(du),
+                                 local_approx);
+
+    // Check we can do derivatives when the components of `u` aren't contiguous
+    // in memory.
+    // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
```

Comment:
you can delete this line when rebasing

Reply:
When I removed this line, clang-tidy reports this error:
```
/__w/spectre/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_PartialDerivatives.cpp:181:16: error: the const qualified variable 'non_contiguous_u' is copy-constructed from a const reference; consider making it a const reference [performance-unnecessary-copy-initialization,-warnings-as-errors]
    const auto non_contiguous_u = get<var_tag>(u);
               ^
```

From this error, I'm thinking that the comment should stay, so I've put it back. Let me know if I'm misunderstanding something!

Reply:
Yes, what you have is correct here @macedo22 The `NOLINT` should be there and the comment right above it explains what is being done (at least to me it's clear)

Reply:
Agreed!

Reply:
Ah sorry I thought the the nolint referred to the decltype syntax you had here before. All good.